x-pack/plugins/actions/server/routes/get_service_now_access_token.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { schema } from '@kbn/config-schema';
+import { IRouter, Logger } from '@kbn/core/server';
+import { ILicenseState } from '../lib';
+import { INTERNAL_BASE_ACTION_API_PATH } from '../../common';
+import { ActionsRequestHandlerContext } from '../types';
+import { verifyAccessAndContext } from './verify_access_and_context';
+import { getAccessToken } from '../builtin_action_types/servicenow/utils';
+import { ActionsConfigurationUtilities } from '../actions_config';
+
+const bodySchema = schema.object({
+  apiUrl: schema.string(),
+  config: schema.object({
+    clientId: schema.string(),
+    jwtKeyId: schema.string(),
+    userIdentifierValue: schema.string(),
+  }),
+  secrets: schema.object({
+    clientSecret: schema.string(),
+    privateKey: schema.string(),
+    privateKeyPassword: schema.maybe(schema.string()),
+  }),
+});
+
+export const getServiceNowAccessToken = (
+  router: IRouter<ActionsRequestHandlerContext>,
+  licenseState: ILicenseState,
+  logger: Logger,
+  configurationUtilities: ActionsConfigurationUtilities
+) => {
+  router.post(
+    {
+      path: `${INTERNAL_BASE_ACTION_API_PATH}/connector/_servicenow_access_token`,
```

Comment:
Is it possible to generalize the endpoint so it can be used by all connectors for future usage?

Reply:
I pushed a commit to refactor some of the utility functions to be more generic.

Reply:
Thanks for the changes! Sorry for the extra work. I thought it was simpler.